pkg/apiserver/apic.go

```diff
@@ -118,13 +164,111 @@ func NewAPIC(config *csconfig.OnlineApiClientCfg, dbClient *database.Client, con
 		Password:       password,
 		UserAgent:      fmt.Sprintf("crowdsec/%s", cwversion.VersionStr()),
 		URL:            apiURL,
+		PapiURL:        PapiURL,
 		VersionPrefix:  "v2",
 		Scenarios:      ret.scenarioList,
 		UpdateScenario: ret.FetchScenariosListFromDB,
 	})
+
 	return ret, err
 }
 
+func (a *apic) SyncDecisions() error {
+	defer types.CatchPanic("lapi/syncDecisionsToCAPI")
+
+	var cache models.AddSignalsRequestItemDecisions
+	ticker := time.NewTicker(a.pushInterval)
+	log.Infof("Start decisions sync to CrowdSec Central API (interval: %s)", PushInterval)
+
+	for {
+		select {
+		case <-a.pushTomb.Dying(): // if one apic routine is dying, do we kill the others?
+			a.pullTomb.Kill(nil)
+			a.metricsTomb.Kill(nil)
+			log.Infof("push tomb is dying, sending cache (%d elements) before exiting", len(cache))
+			if len(cache) == 0 {
+				return nil
+			}
+			go a.SendDeletedDecisions(&cache)
+			return nil
+		case <-ticker.C:
+			if len(cache) > 0 {
+				a.mu.Lock()
+				cacheCopy := cache
+				cache = make([]*models.AddSignalsRequestItemDecisionsItem, 0)
+				a.mu.Unlock()
+				log.Infof("Signal push: %d signals to push", len(cacheCopy))
+				go a.SendDeletedDecisions(&cacheCopy)
+			}
+		case deletedDecisions := <-a.DecisionDeleteChan:
+
+			if a.consoleConfig.ShareManualDecisions != nil && *a.consoleConfig.ShareManualDecisions {
```

Comment:
We need to be careful with this flag and the check overall.

`ShareManualDecisions` configuration option was initially used to send the decisions that a user added manually (`cscli decisions add ..`).
Since we have now the `decisions_poll` configuration option, a user might want to send deleted decisions (to have the synchronization in the console) but not the `decisions` that he added manually.